src/Symfony/Component/DependencyInjection/Compiler/CheckTypeDeclarationsPass.php

```diff
@@ -219,6 +220,10 @@ private function checkType(Definition $checkedDefinition, $value, \ReflectionPar
             return;
         }
 
+        if ('callable' === $type && $value instanceof ServiceClosureArgument) {
```

Comment:
Could you maybe add this case in the previous condition so we only have 1 if for `callable`?

Reply:
I thought about that, but also considered that it might make that clause unreadable. I'll happily add it in though if that's the consensus here.

Reply:
should we accept `\Closure::class` also btw?

Reply:
@shieldo I mean:
```php
if ('callable' === $type) {
    if (\is_array($value)/* etc. */) {
        return;
    }

    if ($value instanceof ServiceClosureArgument) {
        return;
    }
}
```

Reply:
@nicolas-grekas I think that was done in the previous commit https://github.com/symfony/symfony/commit/d6a7fbfbc3c988068754e519ae70ace62a7944a7 unless I'm mistaken?

(Do you think I should follow the suggestion from @fancyweb above?)

Reply:
> should we accept \Closure::class also btw?

I don't think so because is_callable of a \Closure instance returns true so the value is checked and is valid.

Reply:
I mean:
`if (`'callable' === $type || 'Closure' === $type) && $value instanceof ServiceClosureArgument) {`

(about the CS, I'm fine with the original proposal personally.)

Reply:
OK @fancyweb I've updated as per your suggestion - I think that's a pretty happy medium :)

Reply:
@nicolas-grekas I'd agree with that (and would also undo my previous change to make a distinct case in the code for that, as well as providing a new test). @fancyweb Make sense to you?

Reply:
Do it @shieldo, we indeed need to support the \Closure case here.

Reply:
Updated to handle a `\Closure` type hint